qa/rpc-tests/wallet_changeindicator.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python2
+# Copyright (c) 2018 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_true, assert_false, connect_nodes_bi, \
+    initialize_chain_clean, start_nodes, wait_and_assert_operationid_status
+
+from decimal import Decimal
+
+class WalletChangeIndicatorTest (BitcoinTestFramework):
+    # Setup Methods
+    def setup_chain(self):
+        print "Initializing test directory " + self.options.tmpdir
+        initialize_chain_clean(self.options.tmpdir, 2)
+
+    def setup_network(self, split = False):
+        self.nodes = start_nodes(2, self.options.tmpdir)
+        connect_nodes_bi(self.nodes, 0, 1)
+        self.is_network_split = False
+        self.sync_all()
+
+    # Helper Methods
+    def generate_and_sync(self):
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+    # Tests
+    def run_test(self):
+        print "Mining blocks..."
+        self.nodes[0].generate(101)
+
+        taddr = self.nodes[1].getnewaddress()
+        zaddr1 = self.nodes[1].z_getnewaddress()
+        zaddr2 = self.nodes[1].z_getnewaddress()
+
+        self.nodes[0].sendtoaddress(taddr, Decimal('1.0'))
+        self.generate_and_sync()
+
+        # Send 1 ZEC to a zaddr
+        wait_and_assert_operationid_status(self.nodes[1], self.nodes[1].z_sendmany(taddr, [{'address': zaddr1, 'amount': 1.0, 'memo': 'c0ffee01'}], 1, 0))
+        self.generate_and_sync()
+
+        # Check that we have received 1 note which is not change
+        receivedbyaddress = self.nodes[1].z_listreceivedbyaddress(zaddr1, 0)
+        listunspent = self.nodes[1].z_listunspent()
+        assert_equal(1, len(receivedbyaddress), "Should have received 1 note")
+        assert_false(receivedbyaddress[0]['change'], "Note should not be change")
+        assert_equal(1, len(listunspent), "Should have 1 unspent note")
+        assert_false(listunspent[0]['change'], "Unspent note should not be change")
+
+        # Generate some change
+        wait_and_assert_operationid_status(self.nodes[1], self.nodes[1].z_sendmany(zaddr1, [{'address': zaddr2, 'amount': 0.6, 'memo': 'c0ffee02'}], 1, 0))
+        self.generate_and_sync()
+
+        # Check zaddr1 received
+        sortedreceived1 = sorted(self.nodes[1].z_listreceivedbyaddress(zaddr1, 0), key = lambda received: received['amount'])
+        assert_equal(2, len(sortedreceived1), "zaddr1 Should have received 2 notes")
+        assert_equal(Decimal('0.4'), sortedreceived1[0]['amount'].quantize(Decimal('1.0')))
```

Comment:
`.quantize(Decimal('1.0'))` should not be necessary, as the output amounts will already be Decimals, and you are sending the transactions with zero fee. Did you have issues without it?